### PR TITLE
feat: onboard new destination ninetailed

### PR DESCRIPTION
**Pull request by anantjain45823**
## What are the changes introduced in this PR?
Onboard new destination ninetailed.

Write a brief explainer on your code changes.
This destination supports identify, track, and page calls.
Also it supports batching with max batch size as 200

## What is the related Linear task?

Resolves INT-295

## Please explain the objectives of your changes below

Put down any required details on the broader aspect of your changes. If there are any dependent changes, **mandatorily** mention them here

### Any changes to existing capabilities/behaviour, mention the reason & what are the changes ?

N/A

### Any new dependencies introduced with this change?

N/A

### Any new generic utility introduced or modified. Please explain the changes.

N/A

### Any technical or performance related pointers to consider with the change?

N/A

@coderabbitai review

<hr>

### Developer checklist

- [x] My code follows the style guidelines of this project

- [x] **No breaking changes are being introduced.**

- [ ] All related docs linked with the PR?

- [ ] All changes manually tested?

- [ ] Any documentation changes needed with this change?

- [ ] Is the PR limited to 10 file changes?

- [x] Is the PR limited to one linear task?

- [ ] Are relevant unit and component test-cases added?

### Reviewer checklist

- [x] Is the type of change in the PR title appropriate as per the changes?

- [x] Verified that there are no credentials or confidential data exposed with the changes.

## Summary by CodeRabbit

- **New Features**
	- Introduced configuration settings for the Ninetailed destination, including event batch processing and mapping configurations.
- **Tests**
	- Added comprehensive test scenarios for Ninetailed destination integration, covering tracking, page views, user identification, and validation failures.

### CODE REVIEW

**Comment by devops-github-rudderstack** [comment]
Test report for this run is available at: https://test-integrations-dev.s3.amazonaws.com/integrations-test-reports/rudder-transformer/3106/test-report.html

**Comment by saikumarrs** [comment]
@coderabbitai review

File: src/cdk/v2/destinations/ninetailed/utils.js [hunk 1]
```diff
@@ -0,0 +1,38 @@
...
+      typeSpecifcPayload = constructPayload(message, mappingConfig[ConfigCategories.IDENTIFY.name]);
+      break;
+    case 'page':
+      typeSpecifcPayload = constructPayload(message, mappingConfig[ConfigCategories.PAGE.name]);
+      break;
+    default:
+      break;
+  }
+  payload = { ...payload, ...context };
+  return { ...payload, ...typeSpecifcPayload }; // merge base and type-specific payloads;
```
Comment (koladilip) [inline src/cdk/v2/destinations/ninetailed/utils.js:28]:
```suggestion
  return { ...payload, ...context, ...typeSpecifcPayload }; // merge base and type-specific payloads;
```

File: test/integrations/destinations/ninetailed/processor/validation.ts [hunk 2]
```diff
@@ -0,0 +1,180 @@
...
+        status: 200,
+        body: [
+          {
+            error:
+              'Missing required value from "anonymousId": Workflow: procWorkflow, Step: preparePayload, ChildStep: undefined, OriginalError: Missing required value from "anonymousId"',
+            metadata: {
+              destinationId: 'dummyDestId',
+              jobId: '1',
+            },
+            statTags: {
```
Comment (koladilip) [inline test/integrations/destinations/ninetailed/processor/validation.ts:100]: looks like repeated

File: src/cdk/v2/destinations/ninetailed/utils.js (hunk 1 above)
Comment (anantjain45823) [reply src/cdk/v2/destinations/ninetailed/utils.js]: done

File: test/integrations/destinations/ninetailed/processor/validation.ts (hunk 2 above)
Comment (anantjain45823) [reply test/integrations/destinations/ninetailed/processor/validation.ts]: done

File: test/integrations/destinations/ninetailed/processor/identify.ts [hunk 3]
```diff
@@ -0,0 +1,149 @@
...
+    },
+  },
+  {
+    id: 'ninetailed-test-identify-failure-1',
+    name: 'ninetailed',
+    description: 'identify call with no userId available',
+    scenario: 'Framework',
+    successCriteria:
+      'Error should be thrown for required field userId not present and status code should be 200',
+    feature: 'processor',
```
Comment (shrouti1507) [inline test/integrations/destinations/ninetailed/processor/identify.ts:112]: Does this mean there is no way to map non-identified users?

File: test/integrations/destinations/ninetailed/router/data.ts [hunk 4]
```diff
@@ -0,0 +1,252 @@
...
+              ],
+              batched: true,
+              statusCode: 200,
+              destination,
+            },
+          ],
+        },
+      },
+    },
+  },
```
[the remark this answers is not in the document]
Comment (anantjain45823) [reply test/integrations/destinations/ninetailed/router/data.ts]: done

File: test/integrations/destinations/ninetailed/processor/validation.ts [hunk 5]
```diff
@@ -0,0 +1,115 @@
...
+    feature: 'processor',
+    module: 'destination',
+    version: 'v0',
+    input: {
+      request: {
+        body: [
+          {
+            destination,
+            message: {
+              type: 'group',
```
[the remark this answers is not in the document]
Comment (anantjain45823) [reply test/integrations/destinations/ninetailed/processor/validation.ts:73]: done

**Review by koladilip — APPROVED** [review]

**Review by shrouti1507 — APPROVED** [review]

**Review by Gauravudia — APPROVED** [review]

**Pull request merged**